### ISSUE TITLE
Setup time to talk technical details.

### ISSUE THREAD

**Issue by Carreau** [ian-r-rose/jupyterlab-remote-data#2]
Hi all, and @ian-r-rose in particular, 

The couple last week were pretty busy and I haven't moved forward much, though I've been thinking about things. 

First, as I had issue to make this repo to build with my JupyterLab, I made [a fork](https://github.com/Carreau/remote-data-second-test) from start, that use mimerenderer instead of registering directly as Ian did. The main reason was to be able to test the view of remote filesystem when using the file browser. 

So far the Server side content manager works fine, given some heuristic ( a file size for example or a file extensions), it decides wether to send the full content or just a "header" with more information. 
It also change the mimetype so that an extension can react to it. 

There a couple of things I don't really like with this approach, or that I misunderstood:

1) this duplicate all the mimetypes. For every mimetype that can get the full data at once, extension that want to opt-in to streaming need to register a second mimetype. OR I misunderstood how @ian-r-rose code does the dispatch on that. 

2) That does not play well with mimerenderer that register to file extensions, as a file extensions can be both streamed or not depending on the context (for example file size). A 1Mb CSV can likely be sent as is, but not a multi-gigabyte one. 

How to we register things both on frontend and backend ? Or do we just assumes users will register both, as someone using such a content management know what they are doing ? 

Ian, do you have some time Friday or next week to spend 60-90 minutes on Zoom and coding on that with me ?

**Comment by ian-r-rose** [comment]
Hi @Carreau,

Unfortunately, I also have not had a ton of time to make much progress on this, but have also been thinking about it.

Your two points are good ones, and ones that I don't have a solution to right now. I think it will involve some work upstream. In particular, the JupyterLab document manager chooses which viewer to use for a file *before* the file gets loaded to the front-end. So if the backend contents manager is selecting whether to stream or send the full contents, it is already too late. I think, however, it may be possible to patch the document manager so that it selects the viewer once it knows which version to choose.

For registering on the frontend/backend, I also don't have a good solution right now. We could spec out a JSON metadata file somewhere that both the backend and frontend-bundle read?

Friday would work well for me for a technical meetup. It would also be fun to come up to Merced some time :)

**Comment by saulshanabrook** [comment]
If you have a meeting over zoom, I would love to listen in! I am working on implementing data set conversions in JupyterLab and recently talked to @ellisonbg about how he sees this work impacting core (https://github.com/jupyterlab/jupyterlab/issues/5831#issuecomment-453571486).

I would love to get a better sense of the scope of this work so that we can build off of each other.

What if there are different mimetypes defined for the URL vs. the text vs. the parsed data of different formats? Then the conversions between them can be registered as data converters? In this case, a CSV file could start as a `text/csv+url` dataset, which just contains it's URL as its data. This could be converted into a `text/csv+string` which is computed by fetching the URL and reading the response into a string. Finally, it that could be converted into a `application/tabular` mime bundle which would be a fully parsed version of the CSV file. Or a `application/data-model` which could be an instance of the `DataModel` which parses the data on demand.

**Comment by ian-r-rose** [comment]
Yep, you are absolutely invited to whatever we end up scheduling @saulshanabrook!
